blog: introduce SBOMscanner (#384)
kubewarden/kubewarden.io: content/blog/2025/11/expanding-kubewarden-scope.md

---
title: "Expanding Kubewarden Scope"
authors:
  - Flavio Castelli

> date: 2025-11-07

Reviewer: ```suggestion
date: 2025-11-10
```

---

The Kubewarden project was created four years ago at SUSE with the goal of redefining Policy As Code. We built a universal policy engine for Kubernetes and donated it to the CNCF.

When the project started, policies could only be written in Rust and Go. Since then, we've worked to increase flexibility. Today, policies can also be written in other programming languages such as C#, and even JavaScript and TypeScript (stay tuned for the upcoming announcement).

It's also possible to use DSLs like [Rego](https://docs.kubewarden.io/tutorials/writing-policies/rego/intro-rego), and seamlessly [lift and shift](https://docs.kubewarden.io/howtos/gatekeeper-migration) Gatekeeper and Open Policy Agent policies to Kubewarden. The same applies to [CEL](https://docs.kubewarden.io/tutorials/writing-policies/CEL/intro-cel), with the ability to reuse Kubernetes Validating Admission Policies without modification. We even have an [experimental way](https://github.com/kubewarden/kyverno-dsl-policy) to write policies using Kyverno's syntax.

> Supporting different languages hasn't been our only goal. We've also focused on [performance and scalability](https://docs.kubewarden.io/howtos/deploy-at-scale) in collaboration with our community. With [Raw policies](https://docs.kubewarden.io/howtos/raw-policies), we've made it possible to reuse our Policy Server to validate any kind of JSON object, even outside Kubernetes.

Reviewer 2: ```suggestion
Supporting different languages isn't our only goal. We're also focused on [performance and scalability](https://docs.kubewarden.io/howtos/deploy-at-scale) in collaboration with our community. Our [Raw policies](https://docs.kubewarden.io/howtos/raw-policies) make it possible to use our Policy Server to validate any kind of JSON object, even outside Kubernetes.
```

We continue to expand the list of [host capabilities](https://docs.kubewarden.io/reference/spec/host-capabilities/intro-host-capabilities) available to policy authors, adapting to the constantly changing requirements for securing cloud native systems.

> While there have been many improvements, today I'm excited to announce something new: the Kubewarden project is expanding its scope, looking beyond admission policies.

Reviewer 2: ```suggestion
Kubewarden has gone from strength to strength and today I'm excited to announce something new. The Kubewarden project is expanding its scope, looking beyond admission policies.
```

# Announcing SBOMscanner

As mentioned earlier, the requirements to keep cloud native environments secure are constantly growing. One of the most critical aspects is keeping track of vulnerabilities affecting the workloads running inside Kubernetes clusters.

> We took this challenge seriously and set out to find a solution. As a result, today SUSE is making another donation to CNCF, contributing a new project under the Kubewarden umbrella.

Reviewer 2: ```suggestion
We take this challenge seriously and have found a solution. Today SUSE is making another donation to CNCF, contributing a new project under the Kubewarden umbrella.
```

# What makes SBOMscanner different

SBOMscanner is an image scanner designed to integrate seamlessly with Kubernetes. It uses native concepts like Custom Resource Definitions and leverages advanced Kubernetes features such as the [API Aggregation Layer](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/) for performance and scalability.

As the name suggests, SBOMscanner is built around the concept of Software Bill of Materials (SBOM). Images are downloaded once, their SBOMs are created, and then, at regular intervals, they are scanned for vulnerabilities. This approach reduces the time and resources required to keep the ever-growing list of container images scanned.

Currently, SBOMscanner relies on [trivy](https://github.com/aquasecurity/trivy) to scan container images. However, it's designed to support multiple vulnerability scanners. Building SBOMscanner wouldn't have been possible without trivy, as well as CNCF projects like [NATS](https://nats.io/) and [CloudNativePG](https://cloudnative-pg.io/).

# Trying SBOMscanner

> SBOMscanner is not yet production ready, but it's already capable of delivering value, especially when used together with Kubewarden. We've created [this policy](https://github.com/kubewarden/image-cve-policy) that allows you to prevent the usage of vulnerable container images inside your Kubernetes cluster.

Reviewer 2: ```suggestion
SBOMscanner is not yet production-ready, but it's already capable of delivering value, especially when used with Kubewarden. We've created [this policy](https://github.com/kubewarden/image-cve-policy) that allows you to prevent the usage of vulnerable container images inside your Kubernetes cluster.
```
